src/providers/oracle/qgsoracledataitems.cpp

```diff
@@ -135,12 +135,12 @@ QgsOracleConnectionItem::~QgsOracleConnectionItem()
 
 void QgsOracleConnectionItem::stop()
 {
-  if ( mColumnTypeThread )
+  if ( mColumnTypeTask )
   {
-    mColumnTypeThread->stop();
-    mColumnTypeThread->wait();
-    delete mColumnTypeThread;
-    mColumnTypeThread = nullptr;
+    mColumnTypeTask->cancel();
+    mColumnTypeTask->waitForFinished();
```

Comment:
Is this needed? If you disconnected from all its signals then you could just let it bumble away in the background until it finishes on its own and gets deleted by task manager. That'd avoid the potential UI hang here.

Reply:
Right! I modified my PR